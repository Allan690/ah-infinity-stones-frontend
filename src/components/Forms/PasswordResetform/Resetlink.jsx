/* eslint-disable no-useless-escape */
import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import './ResetPassword.scss';
import { passwordReset } from '../../../services/passwordResetmail';
import { requestPasswordResetStart } from '../../../redux/actions/passwordResetAction';


class ResetlinkForm extends React.Component {
    state = {
      email: '',
      error: null,
      success: null,
    }

    handleChange = (e) => {
      const { name, value } = e.target;
      this.setState({ [name]: value });
    };

    handleEmailSubimt = (e) => {
      e.preventDefault();
      const { email } = this.state;
      this.props.requestPasswordResetStart({ email });

      if (!this.validateEmail(email)) {
        this.setState({ error: 'You have entered an invalid email address!' });
        return;
      }
      this.setState({ error: null });
      passwordReset(email).then((response) => {
        this.setState({ success: response.Message });
      }).catch((error) => {
    
      });
      this.setState({ success: null });
    };


    validateEmail(inputText) {
      const mailformat = /^\w+([\.-]?\w+)*@\w+([\.-]?\w+)*(\.\w{2,3})+$/;
      return inputText.match(mailformat);
    }

    render() {
      const { error, success } = this.state;
      return (
        <div>
          <form>
            {error ? <p>{error}</p> : ''}
            {success ? <p>{success}</p> : ''}
            <span className="new_password_label">
                        Enter your email:
            </span>

            <input id="email" type="text" name="email" className="input" value={this.state.email} onChange={this.handleChange} />
            <br />
            <br />

            <br />
            <span>
              <button type="button" className="button-success" onClick={this.handleEmailSubimt}>Submit</button>
            </span>
          </form>

        </div>

      );
    }
}

const mapStateToProps = state => ({
  reset: state.passwordReset,
});

ResetlinkForm.propTypes = {
  success: PropTypes.string,
  error: PropTypes.string,
  requestPasswordResetStart: PropTypes.func.isRequired,
  loading: PropTypes.bool,
};

const actionCreators = { requestPasswordResetStart };


export default connect(mapStateToProps, actionCreators)(ResetlinkForm);
