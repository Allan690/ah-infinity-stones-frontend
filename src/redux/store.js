/* eslint no-underscore-dangle: 0 */
/* eslint-disable no-undef */
import { applyMiddleware, createStore, compose } from 'redux';
import createSagaMiddleware from 'redux-saga';
import allReducers from './reducers';
import rootSaga from './sagas';

// only activate redux devtools on the browser otherwise just compose
const composer = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;
const composeEnhancers = composer;

const middleware = createSagaMiddleware();
const store = createStore(
  allReducers,
  composeEnhancers(applyMiddleware(middleware)),
);

export default store;

middleware.run(rootSaga);
