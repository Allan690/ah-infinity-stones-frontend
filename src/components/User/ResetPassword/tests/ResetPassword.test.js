import React from 'react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import Resetlink from '../../../Forms/PasswordResetform/Resetlink';
import store from '../../../../redux/store';

describe('ResetPassword link component', () => {
  let resetLinkContainer;
  let reduxResetLink;
  beforeEach(() => {
    resetLinkContainer = shallow(<Resetlink className="form_container" />);
    reduxResetLink = mount(
      <Provider store={store}>
        <BrowserRouter>
          <Resetlink />
        </BrowserRouter>
      </Provider>,
    );
  });
  it('form renders successfully', () => {
    expect(resetLinkContainer.exists()).toEqual(true);
  });
  it('Should match snapshots', () => {
    expect(resetLinkContainer).toMatchSnapshot();
  });
  it('Reset link should render with redux store', () => {
    expect(reduxResetLink.exists()).toEqual(true);
  });
  it('renders props correctly', () => {
    const Reset = shallow(<Resetlink name="Resetlink" />);
    expect(Reset.instance().props.name).toBe('Resetlink');
  });
});
