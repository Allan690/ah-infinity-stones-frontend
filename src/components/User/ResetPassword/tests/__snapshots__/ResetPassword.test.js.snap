// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResetPassword link component Should match snapshots 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
